iceoryx_integrationtest/package.xml

```diff
@@ -15,6 +15,7 @@
   <build_depend>ros_testing</build_depend>
 
   <test_depend>ros_testing</test_depend>
+  <test_depend>iceoryx_platform</test_depend>
```

Comment:
This will not work since CMake library does not have an own package.xml file. Therefore colcon will not know where the package is located and cannot build it before the integration tests.
Just adding a package.xml to `iceoryx_hoofs/platform` will also not work since iceoryx_hoofs already contains a package.xml file and having a ROS 2 package inside of another package is not allowed.
The only chance here is to flesh out platform into an own package beside iceoryx_hoofs and posh to make it work.

We should think carefully before doing that to consider aspects of integration etc.

Reply:
The follow up PR #1627 addresses exactly this issue by introducing a `package.xml` for the iceoryx platform.